src/threshold_decryption.rs

```diff
@@ -38,8 +38,13 @@ pub enum Error {
 pub type Result<T> = ::std::result::Result<T, Error>;
 
 /// A Threshold Decryption message.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Rand)]
-pub struct Message(pub DecryptionShare);
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub enum Message {
+    /// A decryption share from a peer.
+    Share(DecryptionShare),
+    /// A message setting nodeId's ciphertext Should only be transmitted locally.
+    Cipher(Ciphertext),
```

Comment:
I might be hacking the handle_message api here. The ciphertext should only be passed internally and not  become network traffic. I think this is enforced through the honeybadger message type -- which intentionally has no definition for a Ciphertext message content, but let me know if this is sketchy or not.

Reply:
Yes, the `Message ` should stay the same.
Instead, the ciphertext should be passed into the constructor of `ThresholdDecryption`.

Reply:
Does that work? Isn't it possible for threshold shares to arrive prior to the ciphertext being sent from the broadcast?

Reply:
That's true, it would move the complexity of storing shares before knowing the ciphertext from `threshold_decryption` to `honey_badger::epoch_state`, so maybe it's a bad idea.

#286 is very much a design issue, too: We need to find the best approach.

But we definitely don't want to send the ciphertext across the network again (that's what has already been done at this point, using the `Broadcast`/RBC algorithm): `Message` won't need to change.

Reply:
By validated do you mean something other than attempting to decrypt after F+1 incoming shares are received?

If not, incoming shares can't be validated without the ciphertext, and a node's share can't be sent without the ciphertext right? So, since shares and ciphertext arrive asynchronously I think the current design is optimal already, as the node's share is sent as soon as the ciphertext arrives plus `try_output` is called as well.

Reply:
> something other than attempting to decrypt

Yes, like the current code already does:
https://github.com/poanetwork/hbbft/blob/2fed8316890f6907d8155d815dd614fdca6978d1/src/threshold_decryption.rs#L169

Exactly, that's only possible with the ciphertext: So currently we validate as soon as we have the ciphertext, and we decrypt as soon as we have ciphertext and _f + 1_ shares.

> the current design is optimal already

Yes, it's fine, given its intended usage inside hbbft.
I just opened #286 because I find it a bit inconsistent how the API of `ThresholdDecryption` differs from the `ThresholdSign` one. I'm not sure about it, but I feel like it would make sense to bring the two in line.